src/mlpack/methods/kmeans/naive_kmeans_impl.hpp

```diff
@@ -38,28 +40,44 @@ double NaiveKMeans<MetricType, MatType>::Iterate(const arma::mat& centroids,
   counts.zeros(centroids.n_cols);
 
   // Find the closest centroid to each point and update the new centroids.
-  for (size_t i = 0; i < dataset.n_cols; i++)
+  // Computed in parallel over the complete dataset
+  #pragma omp parallel
   {
-    // Find the closest centroid to this point.
-    double minDistance = std::numeric_limits<double>::infinity();
-    size_t closestCluster = centroids.n_cols; // Invalid value.
+    // The current state of the K-means is private for each thread
+    arma::mat localCentroids(centroids.n_rows, centroids.n_cols,
+                             arma::fill::zeros);
+    arma::Col<size_t> localCounts(centroids.n_cols, arma::fill::zeros);
 
-    for (size_t j = 0; j < centroids.n_cols; j++)
+    #pragma omp for
+    for (omp_size_t i = 0; i < (omp_size_t) dataset.n_cols; ++i)
     {
-      const double distance = metric.Evaluate(dataset.col(i), centroids.col(j));
+      // Find the closest centroid to this point.
+      double minDistance = std::numeric_limits<double>::infinity();
+      size_t closestCluster = centroids.n_cols; // Invalid value.
 
-      if (distance < minDistance)
+      for (size_t j = 0; j < centroids.n_cols; j++)
       {
-        minDistance = distance;
-        closestCluster = j;
+        const double distance = metric.Evaluate(dataset.col(i),
```

Comment:
Pedantic style issue: when wrapping a line, the next line should be tabbed twice: for more information take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping. This also applies to  ``localCentroids``.